app/models/cart_item.rb

```diff
@@ -4,8 +4,12 @@
 module Shop
   module Models
     class CartItem
-      def initialize(product_id:, quantity: 0)
-        @product = Services::FetchProduct.new.call(product_id)
+      def initialize(product_id:, quantity: 1)
+        @product = begin
+          Services::Products::Fetch.new.call(product_id: product_id)
+        rescue
```

Comment:
you should NEVER rescue everything (rescue without any given error). Rescue everything lead to many hard for debug problems